Tests/Unit/Components/WebService/Soap/ParseHeadersTraitTest.php

```diff
@@ -0,0 +1,91 @@
+<?php
+
+namespace Smartbox\Integration\FrameworkBundle\Tests\Unit\Components\WebService\Soap;
+
+use Smartbox\Integration\FrameworkBundle\Components\WebService\Soap\ParseHeadersTrait;
+use PHPUnit\Framework\TestCase;
+use Symfony\Component\HttpFoundation\Request;
+
+class ParseHeadersTraitTest extends TestCase
+{
+    /**
+     * @var
+     */
+    private $parseTrait;
+
+    /**
+     * @var array
+     */
+    private $httpMethods = [
+        Request::METHOD_POST,
+        Request::METHOD_CONNECT,
+        Request::METHOD_DELETE,
+        Request::METHOD_GET,
+        Request::METHOD_HEAD,
+        Request::METHOD_OPTIONS,
+        Request::METHOD_PURGE,
+        Request::METHOD_PATCH,
+        Request::METHOD_PUT,
+        Request::METHOD_TRACE,
+    ];
+
+    public function setUp()
+    {
+        $this->parseTrait = $this->getMockForTrait(ParseHeadersTrait::class);
+    }
+
+    /**
+     * @group parseSoapHeaders
+     * @dataProvider provideParseableData
+     */
+    public function testParseHeadersToArray($header, $expected)
+    {
+        $result = $this->parseTrait->parseHeadersToArray($header);
+
+        $this->assertTrue(is_array($result), 'The parser should return an array');
+        $this->assertTrue(count($result) > 0, 'The parser should not return an empty array');
+        $this->assertEquals($expected, $result, 'The parser did not returned the same as expected.');
+    }
+
+    /**
+     * @group parseSoapHeaders
+     * @dataProvider provideUnparseableData
+     */
+    public function testUnParseHeadersToArray($data)
+    {
+        $result = $this->parseTrait->parseHeadersToArray($data);
+
+        $this->assertTrue(is_array($result), 'The parser should return an array');
+        $this->assertTrue(0 == count($result), 'The parser should return an empty array');
```

Comment:
There are in PHPUnit asserts to check count and is array. This could be done in this way
```
$this-> assertInternalType('array', $resul);
$this->assertCount(0, $result);
```